src/main/java/com/hazelcast/kubernetes/DnsEndpointResolver.java

```diff
@@ -17,128 +17,113 @@
-                //      Server:         172.30.0.1
-                //      Address:        172.30.0.1#53
-                //
-                //      Name:   u219692-hazelcast.u219692-hazelcast.svc.cluster.local
-                //      Address: 10.1.2.8
-                //      Name:   u219692-hazelcast.u219692-hazelcast.svc.cluster.local
-                //      Address: 10.1.5.28
-                //      Name:   u219692-hazelcast.u219692-hazelcast.svc.cluster.local
-                //      Address: 10.1.9.33
-                SRVRecord srv = (SRVRecord) record;
-                InetAddress[] inetAddress = getAllAddresses(srv);
-                int port = (this.port > 0) ? this.port : getHazelcastPort(srv.getPort());
-                for (InetAddress i : inetAddress) {
-                    Address address = new Address(i, port);
-
-                    // If address is newly discovered and logger is finest, emit log entry
-                    if (addresses.add(address) && logger.isFinestEnabled()) {
-                        logger.finest("Found node service with address: " + address);
-                    }
-                }
-            }
-
-            if (addresses.size() == 0) {
-                logger.warning("Could not find any service for serviceDns '" + serviceDns + "'");
-                return Collections.emptyList();
-            }
+    DnsEndpointResolver(ILogger logger, String serviceDns, int port, int serviceDnsTimeout) {
+        this(logger, serviceDns, port, createDirContext(serviceDnsTimeout));
 
-            return asDiscoveredNodes(addresses);
+    }
 
-        } catch (TextParseException e) {
-            throw new RuntimeException("Could not resolve services via DNS", e);
-        } catch (UnknownHostException e) {
-            throw new RuntimeException("Could not resolve services via DNS", e);
+    @SuppressWarnings("checkstyle:magicnumber")
+    private static DirContext createDirContext(int serviceDnsTimeout) {
+        Hashtable<String, String> env = new Hashtable<String, String>();
+        env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.dns.DnsContextFactory");
+        env.put(Context.PROVIDER_URL, "dns:");
+        env.put("com.sun.jndi.dns.timeout.initial", String.valueOf(serviceDnsTimeout * 1000L));
+        try {
+            return new InitialDirContext(env);
+        } catch (NamingException e) {
+            throw new HazelcastException("Error while initializing DirContext", e);
         }
     }
 
-    private Lookup buildLookup()
-            throws TextParseException, UnknownHostException {
-
-        ExtendedResolver resolver = new ExtendedResolver();
-        resolver.setTimeout(serviceDnsTimeout);
+    List<DiscoveryNode> resolve() {
+        try {
+            return lookup();
+        } catch (NameNotFoundException e) {
+            logger.warning(String.format("DNS lookup for serviceDns '%s' failed: name not found", serviceDns));
```

Comment:
Wdyt about passing exception to `warning` method?

Reply:
I pass it two lines below. Here, I think it's not a good idea, because you actually always get NameNotFoundException when the service is just being started. So, the first POD always prints this warning and we don't want to see tons of exception stack traces.